src/client/context.rs

```diff
@@ -308,6 +309,61 @@ impl Context {
     pub fn set_presence(&self, activity: Option<ActivityData>, status: OnlineStatus) {
         self.shard.set_presence(activity, status);
     }
+
+    /// Gets all emojis for the current application.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if the Application ID is not known.
+    pub async fn get_application_emojis(&self) -> Result<Vec<Emoji>> {
+        self.http.get_application_emojis().await
+    }
+
+    /// Gets information about an application emoji.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if the emoji does not exist.
+    pub async fn get_application_emoji(&self, emoji_id: EmojiId) -> Result<Emoji> {
+        self.http.get_application_emoji(emoji_id).await
+    }
+
+    /// Creates an application emoji with a name and base64-encoded image.
+    ///
+    /// # Errors
+    ///
+    /// See [`Guild::create_emoji`] for information about name and filesize requirements. This
+    /// method will error if said requirements are not met.
+    pub async fn create_application_emoji(&self, name: &str, image: &str) -> Result<Emoji> {
+        let map = json!({
+            "name": name,
+            "image": image,
+        });
```

Comment:
Can you please use a struct with `Deserialize` instead of `json!`? I've been removing this kind of pattern as it allocates and leads to the http interface being restricted to only `Value` instead of the `&impl Serialize` it actually needs.

Reply:
Done.